srtcore/sync.cpp

```diff
@@ -634,17 +644,23 @@ void srt::sync::CThread::create(void *(*start_routine) (void *), void *arg)
     }
 }
 
-#ifdef USE_STDCXX_CHRONO
-template< class Function >
-bool srt::sync::StartThread(CThread& th, Function&& f, void* args, const char* name)
+#endif // !defined(ENABLE_STDCXX_SYNC)
+
+namespace srt {
+namespace sync {
+
+#ifdef ENABLE_STDCXX_SYNC
+typedef void* (&ThreadFunc) (void*);
+template<>
+bool StartThread<Function>(CThread& th, ThreadFunc&& f, void* args, const char* name)
```

Comment:
Did this really compile? Because in this context the `Function` identifier doesn't designate anything.

BTW. unsure why you created a general template with a short specialization. This will result in a linker error instead of being reported in place, when used with wrong function type.

Reply:
Missed when renamed Function to ThreadFunc. Thanks!

>unsure why you created a general template with a short specialization.

What is your suggestion?

Reply:
`ThreadFunc` should be part of the API and this function should be simply a normal function, with `ThreadFunc&&` as an argument.

Reply:
Indeed. Resolved.